backend/routes/upload_reservation_routes.py

```
@@ -46,18 +46,46 @@ def upload_file():
         # Extract form data
-        form_data = {field: request.form.get(field) for field in required_fields}
+        all_required_fields = required_fields + portale_required_fields
+        form_data = {field: request.form.get(field) for field in all_required_fields}
         if any(value is None for value in form_data.values()):
             raise BadRequest("Missing one or more required fields")
 
+        # Validate Portale Alloggi specific fields
+        try:
+            # Validate gender (sesso)
+            if form_data['sesso'] not in ['1', '2']:
+                raise BadRequest("Invalid gender value. Must be 1 (Male) or 2 (Female)")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)